### PR TITLE
AUTH-1309 - Implement V2 of the feedback form

### PR DESCRIPTION
**Pull request by JHjava**
## What?

- Change from a single support page to 3 separate pages. `/contact-us` , '/contact-us-further-information' and '/contact-us-questions'. 
- Implement the different routing which depends on what radio buttons are selected on the '/contact-us' and  '/contact-us-further-information'
- Send the additional information which have been captured in the additional support pages to the zendesk API. 

## Why?

- So we can retrieve more useful feedback from our users

### CODE REVIEW

File: src/components/contact-us/questions/_account-creation-problem-questions.njk [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<h1 class="govuk-heading-l govuk-!-margin-top-0 govuk-!-margin-bottom-3">
+  {{ 'pages.contactUsQuestions.accountCreationProblem.header' | translate }}
+</h1>
+
+<form action="/contact-us-questions" method="post" novalidate>
+
+<input type="hidden" name="_csrf" value="{{csrfToken}}"/>
+<input type="hidden" name="theme" value="{{theme}}"/>
+<input type="hidden" name="subtheme" value="{{subtheme}}"/>
+<input type="hidden" name="backurl" value="{{backurl}}"/>
```
Comment (sanisc) [inline src/components/contact-us/questions/_account-creation-problem-questions.njk:10]: if this is the referrer does it need to be posted back?

File: src/components/contact-us/questions/_account-not-found-questions.njk [hunk 2]
```diff
@@ -0,0 +1,52 @@
+<h1 class="govuk-heading-l govuk-!-margin-top-0 govuk-!-margin-bottom-3">
+  {{ 'pages.contactUsQuestions.accountNotFound.header' | translate }}
+</h1>
+
+<form action="/contact-us-questions" method="post" novalidate>
+
+<input type="hidden" name="_csrf" value="{{csrfToken}}"/>
+<input type="hidden" name="theme" value="{{theme}}"/>
+<input type="hidden" name="subtheme" value="{{subtheme}}"/>
+<input type="hidden" name="backurl" value="{{backurl}}"/>
```
Comment (sanisc) [inline src/components/contact-us/questions/_account-not-found-questions.njk:10]: same here

**Review by sanisc — CHANGES_REQUESTED** [review]

File: src/components/contact-us/contact-us-controller.ts [hunk 3]
```diff
@@ -12,26 +41,79 @@ export function contactUsGet(req: Request, res: Response): void {
...
+  res.redirect(url + "?" + queryParams);
+}
+
+export function contactUsQuestionsGet(req: Request, res: Response): void {
+  let pageTitle = themeToPageTitle[req.query.theme as string];
+  if (
+    req.query.subtheme === ZENDESK_THEMES.SOMETHING_ELSE &&
+    req.query.theme === ZENDESK_THEMES.ACCOUNT_CREATION
+  ) {
+    pageTitle = somethingELseSubThemeToPageTitle[req.query.theme as string];
```
[the remark this answers is not in the document]
Comment (JHjava) [reply src/components/contact-us/contact-us-controller.ts]: Good spot. Done

File: src/components/contact-us/contact-us-controller.ts [hunk 4]
```diff
@@ -12,26 +41,79 @@ export function contactUsGet(req: Request, res: Response): void {
...
+  } else {
```
[the remark this answers is not in the document]
Comment (JHjava) [reply src/components/contact-us/contact-us-controller.ts]: Done

File: src/components/contact-us/questions/_account-creation-problem-questions.njk (hunk 1 above)
Comment (JHjava) [reply src/components/contact-us/questions/_account-creation-problem-questions.njk:10]: Added the `backurl` as a hidden field here so when the page fails validation it still knows where the previous page was and can go back there.

File: src/components/contact-us/contact-us-controller.ts [hunk 5]
```diff
@@ -12,26 +41,79 @@ export function contactUsGet(req: Request, res: Response): void {
   return res.render("contact-us/index-public-contact-us.njk");
 }
 
-export function contactUsSubmitSuccessGet(req: Request, res: Response): void {
-  res.render("contact-us/index-submit-success.njk");
+export function contactUsFormPost(req: Request, res: Response): void {
+  let url = PATH_NAMES.CONTACT_US_QUESTIONS;
+  const queryParams = new URLSearchParams({ theme: req.body.theme }).toString();
+  if (
+    req.body.theme === ZENDESK_THEMES.ACCOUNT_CREATION ||
```
[the remark this answers is not in the document]
Comment (JHjava) [reply src/components/contact-us/contact-us-controller.ts]: Done

**Review by sanisc — APPROVED** [review]

**Pull request merged**